src/lib/pci_virtio_net_vpnkit.c

```diff
@@ -93,13 +95,21 @@ struct msg_init {
 } __packed;
 
 #define CMD_ETHERNET 1
-struct msg_common {
-	uint8_t command;
-} __packed;
+#define CMD_PREFERRED_IPV4 8
+
+#define RESP_VIF 1
+#define RESP_DISCONNECT 2
 
 struct msg_ethernet {
 	uint8_t command; /* CMD_ETHERNET */
 	char uuid[36];
+	uint32_t unused;
+} __packed;
+
+struct msg_preferred_ipv4 {
```

Comment:
Are the messages always the same size? Looks like `msg_ethernet` and `msg_preferred_ipv4` are. If so, a `union` will be a better structure for this.